sdk/test/tutorials/createPortfolio.test.ts

```diff
@@ -1,40 +1,716 @@
+// Test specific libraries
+import mlog from 'mocha-logger';
+import uuidv4 from 'uuid/v4';
+import moment, { Moment } from 'moment';
+import assert, { AssertionError } from 'assert';
+
 // Require the LUSID SDK and libraries
 import {
-  CreateTransactionPortfolioRequest, Portfolio } from "../../api";
-
+  CreateTransactionPortfolioRequest,
+  CreatePropertyDefinitionRequest,
+  Portfolio
+} from "../../api";
 import { client } from './clientBuilder'
+import { Transactions } from '../../client/lusidTools'
+import {
+  PropertyDefinition,
+  LusidProblemDetails,
+  DeletedEntityResponse,
+  ResourceId,
+  PropertyValue,
+  Property,
+  PortfolioProperties,
+  UpsertInstrumentsResponse,
+  InstrumentDefinition,
+  InstrumentIdValue,
+  UpsertPortfolioTransactionsResponse,
+  VersionedResourceListOfTransaction,
+  TransactionRequest,
+  PerpetualProperty,
+  ResourceListOfScopeDefinition,
+  ResourceListOfPortfolio
+} from "../../model/models";
+
+// Lusid method handling libraries
 import { IncomingMessage } from "http";
-import { LusidProblemDetails } from "../../model/models";
-const uuid4 = require('uuid/v4')
 
+// polyfill for Object.fromEntries until Typescript has ES2019 included
+const _ObjectFromEntries = ( iter ): Object => {
+  const obj = {};
+
+  for (const pair of iter) {
+    if (Object(pair) !== pair) {
+      throw new TypeError('iterable for fromEntries should yield objects');
+    }
+
+    // Consistency with Map: contract is that entry has "0" and "1" keys, not
+    // that it is an array or iterable.
+
+    const { '0': key, '1': val } = pair;
+
+    Object.defineProperty(obj, key, {
+      configurable: true,
+      enumerable: true,
+      writable: true,
+      value: val,
+    });
+  }
+
+  return obj;
+}
+
+const createTransactionPortfolio = (
+  {
+    scope
+  }: {
+    scope: string,
+  }
+) :Promise<{
+  portfolio:Portfolio;
+  createRequest:CreateTransactionPortfolioRequest}
+> => {
+
+  return new Promise((resolve, reject) => {
+
+    const
+      createRequest = Object.assign(
+        new CreateTransactionPortfolioRequest(),
+        {
+          displayName: "UK Equities",
+          code: "UKEQTY",
+          baseCurrency: "GBP",
+          created: moment([2018, 1, 1, 0, 0, 0]).utc()
+        }
+      );
+
+    client.api.transactionPortfolios.createPortfolio(
+      scope,
+      createRequest
+    )
+    .then((res: {response: IncomingMessage; body: Portfolio}) => {
+      resolve( { portfolio: res.body, createRequest } )
+    })
+    .catch((err: {response: IncomingMessage; body: LusidProblemDetails}) => reject(err))
+
+  } );
+}
```

Comment:
Seeing some duplication here between this function and the one in the getholdings tests